contracts/ZkSync_SpokePool.sol

```diff
@@ -166,7 +166,13 @@ contract ZkSync_SpokePool is SpokePool, CircleCCTPAdapter {
                 zkUSDCBridge.withdraw(withdrawalRecipient, l2TokenAddress, amountToReturn);
             }
         } else {
-            zkErc20Bridge.withdraw(withdrawalRecipient, l2TokenAddress, amountToReturn);
+            // Elastic chain bridged tokens advertise custom bridge interface via the l2Bridge() getter.
+            address l2Bridge = IZkErc20(l2TokenAddress).l2Bridge();
+            if (l2Bridge != address(0)) {
+                ZkBridgeLike(l2Bridge).withdraw(withdrawalRecipient, l2TokenAddress, amountToReturn);
```

Comment:
tbd catch andrevert to zkErc20Bridge? These tokens are managed by governance, so there's some control over which interfaces the ERC20s implement.